### PR TITLE
Feat: Vibes console app

### PR DESCRIPTION
**Pull request by josenriagu**
## Description

- vibes console app v1

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: ui/apps/vibes-console/package.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    "i18next-fetch-backend": "5.0.2",
+    "i18next-localstorage-backend": "4.1.1",
+    "react-i18next": "12.2.0",
+    "@tanstack/react-query": "4.29.3"
+  },
+  "devDependencies": {
+    "@akashaorg/typings": "*",
+    "@types/react": "18.0.33",
+    "@types/react-dom": "18.0.11",
+    "@types/react-router-dom": "5.3.3",
```
Comment (quininez) [inline ui/apps/vibes-console/package.json]: why do you need types for react-router-dom?

File: ui/apps/vibes-console/src/components/app-routes/index.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+const applicationsRoute = new Route({
+  getParentRoute: () => rootRoute,
+  path: '/applications-center',
+  component: () => {
+    return <Applications />;
+  },
+});
+
+const becomeModeratorRoute = new Route({
+  getParentRoute: () => rootRoute,
```
Comment (quininez) [inline ui/apps/vibes-console/src/components/app-routes/index.tsx:20]: here the parent route is not the rootRoute, it's the applicationsRoute, also the path can be changed to reflect that

File: ui/apps/vibes-console/src/components/app-routes/index.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+});
+
+const routeTree = rootRoute.addChildren([
+  applicationsRoute.addChildren([becomeModeratorRoute]),
+  dashboardRoute,
+  settingsRoute,
+]);
+
+// @todo: update to use type from typings package
+export const createRouter = ({ baseRouteName }: { baseRouteName: string }) =>
```
Comment (quininez) [inline ui/apps/vibes-console/src/components/app-routes/index.tsx]: you can use the type from typings now

File: ui/apps/vibes-console/src/pages/applications.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+export const Applications: React.FC<unknown> = () => {
+  const { t } = useTranslation('vibes-console');
+  const { getRoutingPlugin } = useRootComponentProps();
+
+  const navigateTo = getRoutingPlugin().navigateTo;
+
+  const isModerator = false;
+
+  const handleCtaButtonClick = () => {
+    navigateTo?.({
```
Comment (quininez) [inline ui/apps/vibes-console/src/pages/applications.tsx]: if you are using tanstack-router in this app, then also switch to useNavigate from that library for internal navigation

File: ui/apps/vibes-console/src/pages/become-moderator.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  const [footerChecked, setFooterChecked] = useState(false);
+
+  const { t } = useTranslation('vibes-console');
+  const { getRoutingPlugin } = useRootComponentProps();
+  const navigateTo = getRoutingPlugin().navigateTo;
+
+  const handleLinkClick = (link: string) => {
+    if (link.includes('values')) {
+      navigateTo({
+        appName: '@akashaorg/app-vibes',
```
Comment (quininez) [inline ui/apps/vibes-console/src/pages/become-moderator.tsx:22]: also here

File: ui/apps/vibes/src/components/app-routes/index.tsx [hunk 6]
```diff
@@ -84,13 +84,9 @@ const reportItemRoute = new Route({
...
 
 const routeTree = rootRoute.addChildren([
-  overviewRoute,
-  moderationValueRoute,
-  moderatorsRoute,
-  viewModeratorRoute,
-  historyRoute,
-  historyItemRoute,
-  reportItemRoute,
+  overviewRoute.addChildren([moderationValueRoute]),
```
Comment (quininez) [inline ui/apps/vibes/src/components/app-routes/index.tsx:87]: where you have routes as children of other routes, remember to also provide the parent route, ex: for moderationValueRoute,  getParentRoute: () => overviewRoute,  and also adjust the path

File: ui/apps/vibes/src/pages/overview.tsx [hunk 7]
```diff
@@ -23,10 +23,17 @@ export const Overview: React.FC<OverviewPageProps> = props => {
 
   const navigateTo = getRoutingPlugin().navigateTo;
 
+  const handleClickApply = () => {
+    navigateTo({
```
Comment (quininez) [inline ui/apps/vibes/src/pages/overview.tsx:29]: use useNavigate() from tanstack

File: ui/apps/vibes/src/pages/overview.tsx [hunk 8]
```diff
@@ -23,10 +23,17 @@ export const Overview: React.FC<OverviewPageProps> = props => {
...
+      appName: '@akashaorg/app-vibes-console',
+      getNavigationUrl: (routes: Record<string, string>) => routes['Become Moderator'],
+    });
+  };
+
   const handleCodeOfConductClick = () => {
     navigateTo({
       appName: '@akashaorg/app-legal',
-      getNavigationUrl: routes => routes.codeOfConduct,
+      getNavigationUrl: (routes: Record<string, string>) => routes.codeOfConduct,
```
Comment (quininez) [inline ui/apps/vibes/src/pages/overview.tsx:38]: same here

File: ui/widgets/sidebar/src/components/list-sidebar-apps.tsx [hunk 9]
```diff
@@ -17,6 +17,9 @@ export type ListSidebarAppsProps = {
   onClickMenuItem?: (menuItem: IMenuItem, isMobile?: boolean) => void;
 };
 
+const getIsActiveMenu = (appName: string) =>
+  location.pathname.split('/')[2] === appName.split('/')[1];
```
Comment (quininez) [inline ui/widgets/sidebar/src/components/list-sidebar-apps.tsx]: why not keep using location.pathname.includes(app.name) to check this? this split method seems more vulnerable to changes

**Review by quininez — CHANGES_REQUESTED** [review]

File: ui/apps/vibes-console/package.json (hunk 1 above)
Comment (josenriagu) [reply ui/apps/vibes-console/package.json]: good catch, remnant from before tanstack was implemented, can be removed now

File: ui/apps/vibes-console/src/pages/become-moderator.tsx (hunk 5 above)
Comment (josenriagu) [reply ui/apps/vibes-console/src/pages/become-moderator.tsx:22]: This is not internal navigation tho. it leads to vibes app instead.

Comment (quininez) [reply ui/apps/vibes-console/src/pages/become-moderator.tsx:22]: ok, only where you have internal navigation

**Review by quininez — APPROVED** [review]

**Pull request merged**